src/utils/i18n.ts

```diff
@@ -1,30 +1,61 @@
 import i18n from 'i18next'
-import { initReactI18next } from 'react-i18next'
+import { initReactI18next, useTranslation } from 'react-i18next'
 import en from '../locales/en.json'
 import zh from '../locales/zh.json'
 import { storeCachedData, fetchCachedData } from './cache'
 import { AppCachedKeys } from '../constants/cache'
 
+export type LanuageType = 'en' | 'zh'
+
+export type I18nType = typeof i18n
+
+export type I18nInfoType = {
+  i18n: I18nType
+  currentLanguage: LanuageType
+  setLanguage: (lan: LanuageType) => void
+  toggleLanguage: () => void
+}
+
+const getDefaultLanguage = () => fetchCachedData<LanuageType>(AppCachedKeys.AppLanguage) ?? 'en'
+
 i18n.use(initReactI18next).init({
   resources: {
     en,
     zh,
   },
-  fallbackLng: 'en',
+  fallbackLng: getDefaultLanguage(),
   interpolation: {
     escapeValue: false,
   },
 })
 
-export const changeLanguage = (lan: 'en' | 'zh') => {
-  if (lan.indexOf('zh') !== -1) {
-    i18n.changeLanguage('zh')
-  } else {
-    i18n.changeLanguage('en')
+export function useI18n(): I18nInfoType {
+  const { t } = useTranslation()
```

Comment:
`i18n` can be imported by the hook, the custom hook `useI18n` does the same work of `useTranslation`

Ref:
- https://react.i18next.com/latest/usetranslation-hook